docs/documentation/head/connect.md

```diff
@@ -117,7 +117,7 @@ Connection conn = DriverManager.getConnection(url);
 
 * **sslhostnameverifier** = String
 
-	Class name of hostname verifier. Defaults to using `org.postgresql.ssl.jdbc4.LibPQFactory.verify()`
```

Comment:
Does that mean the code would fail if someone uses `sslhostnameverifier=org.postgresql.ssl.jdbc4.LibPQFactory.verify()` explicitly?

Reply:
So this comment is a bit confusing.  You set the class name, and it uses the <class>#verify method.  It may be better to update this comment to state that it "Defaults to `org.postgresql.ssl.jdbc4.LibPQFactory`, but that is also a bit confusing because setting a custom sslhostnameverifier changes behavior such that hostnames are verified regardless of sslmode (as I stated in the original bug, the interactions between the ssl properties are confusing and not very well documented).

Previously, if one had tried to explicitly set sslhostnameverifier to org.postgresql.ssl.jdbc4.LibPQFactory it would cause very strange behavior (as LibPQFactory would be instantiated and set up it's trust and key maanger regardless of what sslsocketfactory was used).  Now, setting sslhostnameverifier=org.postgresql.ssl.jdbc4.LibPQVerifier will work just fine (and will basically force enable hostname verification even if sslmode isn't verify-full), and is consistent with existing behavior.  If anybody had previously been using a url with sslhostnameverifier=org.postgresql.ssl.jdbc4.LibPQFactory, that will break, but it probably wasn't doing what they expected in the first place.

The main issue that I'm trying to solve is that there is no way to use a custom sslsocketfactory with the default sslhostnameverifier.  Setting sslmode=verify-full and sslsocketfactory=<something> doesn't actually verify hostnames, and as stated above, using a custom sslsocketfactory with LibPQFactory as the sslhostnameverifier will cause strange behavior in the best case, and in most cases is simply impossible.

Reply:
I think @vlsi is trying to determine if this is a breaking change? By changing the name of the sslhostnameverifier any existing code might break.

Reply:
Ya, as is, this could potentially break some existing usage.  One option would be to break out the verification logic into a standalone class, and then use it from the LibPQFactory such that the existing behavior is kept, but I would still be concerned about the behavior of these parameters being very counter-intuitive.